api/envoy/extensions/common/matching/v3/extension_matcher.proto

```diff
@@ -37,3 +37,13 @@ message ExtensionWithMatcher {
   config.core.v3.TypedExtensionConfig extension_config = 2
       [(validate.rules).message = {required: true}];
 }
+
+// Extra settings on a per virtualhost/route/weighted-cluster level.
+message ExtensionWithMatcherPerRoute {
+  // Matcher override.
+  oneof override {
```

Comment:
we only have one option here, which is xds_matcher,
so why we need oneof?

Reply:
I was following the structure of some other per route configs. It doesn't appear necessary in this case.